brainiak/utils/utils.py

```diff
@@ -100,3 +103,164 @@ def fast_inv(a):
+
+        include_orth: Boollean, whether to include "orthogonal" regressors in
+            the design matrix which are usually head motion parameters. All
+            the columns of design matrix are still going to be read in, but
+            the attribute cols_used will reflect whether these orthogonal
+            regressors are to be included for furhter analysis.
+
+        include_pols: Boollean, whether to include polynomial regressors in
+            the design matrix which are used to capture slow drift of signals.
+            This will be reflected in the indices in the attribute cols_used.
+
+        Attributes
+        ----------
+
+        design: 2d array. The design matrix read in from the csv file.
+
+        n_col: number of total columns in the design matrix.
+
+        column_types: 1d array. the types of each column in the design matrix.
+            0 for orthogonal regressors (usually head motion parameters),
+            -1 for polynomial basis (capturing slow drift of signals),
+            values > 0 for stimulus conditions
+
+        n_basis: scalar. The number of polynomial bases in the designn matrix.
+
+        n_stim: scalar. The number of stimulus conditions.
+
+        n_orth: scalar. The number of orthogoanal regressors (usually head
+            motions)
+
+        StimLabels: list. The names of each column in the design matrix.
+    """
+    def __init__(self, fname=None, include_orth=False, include_pols=False):
+        if fname is None:
+            # fname is the name of the file to read in the design matrix
+            self.design = np.zeros([0, 0])
+            self.n_col = 0
+            # number of columns (conditions) in the design matrix
+            self.column_types = np.ones(0)
+            self.n_basis = 0
+            self.n_stim = 0
+            self.n_orth = 0
+            self.StimLabels = []
+        else:
+            # isAFNI = re.match(r'.+[.](1D|1d|txt)$', fname)
+            filename, ext = os.path.splitext(fname)
+            # We assume all AFNI 1D files have extension of 1D or 1d or txt
+            if ext in ['.1D', '.1d', '.txt']:
+                self.readAFNI(fname=fname)
+
+        self.include_orth = include_orth
+        self.include_pols = include_pols
+        self.cols_used = np.where(self.column_types == 1)[0]
+        if self.include_orth:
+            self.cols_used = np.sort(
+                np.append(self.cols_used, np.where(self.column_types == 0)[0]))
+        if self.include_pols:
+            self.cols_used = np.sort(np.append(
+                self.cols_used, np.where(self.column_types == -1)[0]))
+        self.design_used = self.design[:, self.cols_used]
+        if not self.include_pols:
+            # baseline is not included, then we add a column of all 1's
+            self.design_used = np.insert(self.design_used, 0, 1, axis=1)
+        self.n_TR = np.size(self.design_used, axis=0)
+
+    def readAFNI(self, fname):
```

Comment:
Method names should be lowercase, with optional underscores for readability:
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables